internal/kafka/test/helper.go

```diff
@@ -60,12 +60,26 @@ func NewKafkaHelper(t *testing.T, server *httptest.Server) (*test.Helper, *publi
 
 func NewKafkaHelperWithHooks(t *testing.T, server *httptest.Server, configurationHook interface{}) (*test.Helper, *public.APIClient, func()) {
 	h, teardown := test.NewHelperWithHooks(t, server, configurationHook, kafka.ConfigProviders(), di.ProvideValue(environments.BeforeCreateServicesHook{
-		Func: func(dataplaneClusterConfig *config.DataplaneClusterConfig, kafkaConfig *config.KafkaConfig, observabilityConfiguration *observatorium.ObservabilityConfiguration, kasFleetshardConfig *config.KasFleetshardConfig) {
+		Func: func(dataplaneClusterConfig *config.DataplaneClusterConfig, kafkaConfig *config.KafkaConfig, observabilityConfiguration *observatorium.ObservabilityConfiguration, kasFleetshardConfig *config.KasFleetshardConfig, providerConfig *config.ProviderConfig) {
 			kafkaConfig.KafkaLifespan.EnableDeletionOfExpiredKafka = true
 			observabilityConfiguration.EnableMock = true
 			dataplaneClusterConfig.DataPlaneClusterScalingType = config.NoScaling // disable scaling by default as it will be activated in specific tests
 			dataplaneClusterConfig.RawKubernetesConfig = nil                      // disable applying resources for standalone clusters
 			dataplaneClusterConfig.EnableDynamicScaleUpManagerScaleUpTrigger = false
+
+			// remove the gcp provider to avoid integration tests creating instances
```

Comment:
Maybe instead of removing the gcp provider we could set completely the providers value totally ignoring the originally read value from the file? In that case we would have full control of the providers we want by default, totally ignoring the provider configuration file for our integration tests.
cc @machi1990 to have another view around this.

Additionally, did you verify this indeeds changes the original value and not a copy?

Reply:
> Maybe instead of removing the gcp provider we could set completely the providers value totally ignoring the originally read value from the file? In that case we would have full control of the providers we want by default, totally ignoring the provider configuration file for our integration tests.

+1. That way we won't have to ask ourselves the same questions when we support other cloud providers

Reply:
I see this has been updated to include aws but it is still reading the value from the configuration file instead of hardcoding the values. The best is probably to completely hardcode the provider information

Reply:
This is done now, the aws provider is hardcoded.